tests/benchmark/benchmark_gptq.py

```diff
@@ -0,0 +1,232 @@
+    end_event = torch.cuda.Event(enable_timing=True)
+
+    torch.cuda.reset_peak_memory_stats(device)
+    torch.cuda.empty_cache()
+    torch.cuda.synchronize()
+
+    start_event.record()
+    for _ in tqdm(range(num_batches)):
+        if is_decoder:
+            _ = model.generate(input_ids, attention_mask=masks, generation_config=generation_config)
+        else:
+            _ = model(input_ids, masks)
+    end_event.record()
+    torch.cuda.synchronize()
+    max_memory = torch.cuda.max_memory_allocated(device)
+
+    return start_event.elapsed_time(end_event) / num_batches, max_memory
+
+
+def benchmark(model, input_ids: torch.Tensor, masks: torch.Tensor, num_batches: int, is_decoder: bool, new_tokens: int, pad_token_id: int):
+    # Warmup
+    if is_decoder:
+        gen_config = GenerationConfig(
+            max_new_tokens=new_tokens,
+            min_new_tokens=new_tokens,
+            use_cache=True,
+            pad_token_id=pad_token_id,
+            num_beams=1,
+            do_sample=False,
+        )
+        _ = model.generate(input_ids, attention_mask=masks, generation_config=gen_config)
+        torch.cuda.synchronize()
+    else:
+        _ = model(input_ids, masks)
+        torch.cuda.synchronize()
+
+    # Benchmark
+    if is_decoder:
+        total_time, max_mem = timing_cuda(model, num_batches, input_ids, masks, is_decoder, gen_config)
+    else:
+        total_time, max_mem = timing_cuda(model, num_batches, input_ids, masks, is_decoder)
+
+    return total_time, max_mem
+
+
+parser = get_parser()
+args = parser.parse_args()
+
+if args.sweep:
+    batch_sizes = [1, 4, 16, 32]
+    prompt_lengths = [512]
+    new_tokens = [512]
+else:
+    batch_sizes = args.batch_size
+    prompt_lengths = args.prompt_length
+    new_tokens = args.new_tokens
+
+
+if not torch.cuda.is_available():
+    raise ValueError("A cuda device is necessary to benchmark GPTQ.")
+
+device = torch.device("cuda:0")
+tokenizer = AutoTokenizer.from_pretrained(args.model)
+
+if not hasattr(tokenizer, "pad_token") or tokenizer.pad_token is None:
+    tokenizer.pad_token = tokenizer.eos_token
+
+if args.task:
+    task = args.task
+else:
+    task = TasksManager.infer_task_from_model(args.model)
+
+if task == "text-generation":
+    autoclass = AutoModelForCausalLM
+elif task == "text2text-generation":
+    autoclass = AutoModelForSeq2SeqLM
+else:
+    autoclass = AutoModel
+
+if task in ["text-generation", "text2text-generation"]:
+    is_decoder = True
+else:
+    is_decoder = False
+
+if args.gptq:
+    if not args.gptq_model:
+        raise ValueError("The argument --gptq-model needs to be provided when benchmarking GPTQ.")
+    
+    with init_empty_weights():
+        empty_model = AutoModelForCausalLM.from_pretrained(args.model, torch_dtype=torch.float16)
+    empty_model.tie_weights()
+    model = load_quantized_model(empty_model, save_folder=args.gptq_model, state_dict_name="model.safetensors", device_map="auto")
```

Comment:
In this function `disable_exllama` is set to False by default, maybe good to have the choice to disable it ?

Reply:
Yes I'll add both! Also no act-order / act-order.